lib/rpush/daemon/delivery.rb

```diff
@@ -20,7 +20,12 @@ def mark_retryable_exponential(notification)
       end
 
       def mark_batch_retryable(deliver_after, error)
-        log_warn("Will retry #{@batch.notifications.size} notifications after #{deliver_after.strftime('%Y-%m-%d %H:%M:%S')} due to error (#{error.class.name}, #{error.message})")
+        total_count = @batch.notifications.size
+        retry_count = total_count - @batch.delivered.size - @batch.failed.size
```

Comment:
Ok, so now I'm a little concerned that we're duplicating the calculation of what will and will not be retried. I think we have to move the logging down to `#mark_all_retryable` in order to avoid that.

Reply:
We don't have such calculation inside `Rpush::Daemon::Batch` so it is not a duplication or I'm missing something but still it makes some sense because it more looks like the `Rpush::Daemon::Batch` responsibility than `Rpush::Daemon::Delivery`

Reply:
I'm referring to https://github.com/rpush/rpush/pull/567/files#diff-805216e5e54345d16321481b20c83391R30

This is duplicating the knowledge that delivered and failed notifications won't be included. I'd prefer only one method know about that detail.

Reply:
I need to know the exact number for the log message and if we remove such calculation, it will be implemented as kind of counter that I'll increment here during iteration through notifications https://github.com/rpush/rpush/pull/567/files#diff-805216e5e54345d16321481b20c83391R30

But as for me this incremental counter will look strange because we already know the total amount for each kind

Reply:
That's fine, I'd just like to keep them as close together as possible. If you don't see a good way to do that, so be it.